discord/ext/commands/context.py

```diff
@@ -452,6 +452,63 @@ def me(self) -> Union[Member, ClientUser]:
         # bot.user will never be None at this point.
         return self.guild.me if self.guild is not None else self.bot.user  # type: ignore
 
+    @discord.utils.cached_property
+    def permissions(self) -> Permissions:
+        """:class:`.Permissions`: Returns the resolved permissions for the invoking user in this channel.
+        Shorthand for :meth:`.abc.GuildChannel.permissions_for` or :attr:`.Interaction.permissions`.
+
+        .. versionadded:: 2.0
+        """
+        if self.channel.type == ChannelType.private:
+            return Permissions._dm_permissions()
+        if not self.interaction:
+            # channel and author will always match relevant types here
+            return self.channel.permissions_for(self.author)  # type: ignore
+        return self.interaction.permissions
+
+    @discord.utils.cached_property
+    def bot_permissions(self) -> Permissions:
+        """:class:`.Permissions`: Returns the resolved permissions for the bot in this channel.
+
+        .. note::
+
+            For interaction-based commands, this will reflect the effective permissions
+            for :class:`Context` calls, which may differ from calls through
+            other :class:`.abc.Messageable` endpoints, like :attr:`channel`.
+
+            Notably, sending messages, embedding links, and attaching files are always
+            permitted, while reading messages might not be. Mass mentioning and using
+            external emojis and stickers also inherits their permissions from the
+            default guild role.
```

Comment:
You can probably not use the admonition here.

Reply:
Do you mean that I should remove the `.. note::` directive and leave the content of the directive in the docstring, or rather that I should remove the entire block?

Reply:
I meant removing the admonition not the content